### DISCUSSION TITLE
Android notification not working

### DISCUSSION THREAD

**Discussion by InfinityLoop1308** [tutao/tutanota#4423] [Bugs]
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Bug in mobile app**

**Describe the bug**
Android notification not working. Battery optimization is disabled. Can not receive any notifications once you close the app.
When reopen the app, it shows "offline" and reconnects in a few seconds.

**To Reproduce**
1. Download, install, login
2. close the app
3. no notifications, not any.

**Expected behavior**
Notification works properly

**Screenshots**

**Smartphone (please complete the following information):**
 - OS: LineageOS 18.1
 - Version 3.98.15

**Additional context**
This bug has existed for a long time.
I have submitted this issue in January(#3796), and make a fix(#3801). After that, I waited for months for the merge. However, they are all closed for no reason.
I really think Tutanota is an excellent app, but this bug make it totally unusable. I believe I am not the only one who have this problem - there must be others encountered this problem and just leave Tutanota silently.
To make Tutanota better, please, don't ignore this issue and pretend as it never occurs. I am willing to provide any information needed ASAP.

**Comment by charlag** [comment]
Hello, I converted this to a discussion.

We still can't reproduce it and also you don't understand why your fix works. This is the reason the PR is closed. We can't just blindly apply patches.

**Reply by InfinityLoop1308** [reply]
> Hello, I converted this to a discussion.
> 
> We still can't reproduce it and also you don't understand why your fix works. This is the reason the PR is closed. We can't just blindly apply patches.

I understand that you can't blindly apply patches. However, it indeed exists on my device, and I want it to be merged. Since it can be reproduced on my device, I think it is a good idea to begin with finding out what happened on my device. I can provide related info to support you.

**Reply by InfinityLoop1308** [reply]
Another point: is it tested under LineageOS? If no, that may be a bug caused by the OS - it may have some OS specific behavior, e.g. some behavior make Tutanota thinks there is no network connection and makes the reconnecting not work

**Reply by charlag** [reply]
You can definitely start investigating, it doesn't have to be merged for that, you just need to build the client and maybe add some logging to see the difference.

It might very well be LineageOS bug.

**Reply by InfinityLoop1308** [reply]
So what I need to do for it to be merged? Sorry, but I still feel confused.

**Reply by charlag** [reply]
Find out why it doesn't work so that we can reproduce it and figure out how to fix it (or be sure that your fix does what is intended)

**Reply by InfinityLoop1308** [reply]
OK, that's clear enough. I will update in this discussion when I found out the reason.

**Comment by nach0** [comment]
I just want to join and say that I have the same (or a very similar) problem. I rarely receive notifications on my phone, but it sometimes works (I haven't checked under which conditions it works).

I am running iodéOS (based on LineageOS) on a FairPhone 4. Tutanota app version is 3.118.2, but this has been happening for several months (more than 1 year?).

I have seen that there were similar issues reported on GrapheneOS and LineageOS (3051 and 3986), but they are already marked as closed. There is something not working properly on non-standard Android OSs.

**Reply by charlag** [reply]
Even among our devs there are those who use non-standard OSes, I don't think that's the reason

**Comment by BradB58** [comment]
Hello, I am yet another non-standard Android OS user checking in with this issue. It has been a problem for at least a couple of years now. It sure seems to me that the notification system is going to sleep. If you open up the app, notifications will work for a few hours, then stop until you open the app again. I have battery optimization off, however I was wondering why there is not a persistent notification to keep Tuta awake as many other apps use when not using Google FCM.  I use Telegram FOSS and it requires both battery optimization to be off AND a persistent notification in order to work.

Device: Google Pixel 4a (5G)
OS: CalyxOS version 5.11.3
Android Version: 14

**Reply by charlag** [reply]
Hey, it's really surprising to hear that.
Tuta is not a messenger and we don't need need the app to be constantly running. There's a job scheduled every 15 min (or so) that should check notifications. I would look into logs to see if that is the case.

**Comment by paridhi089** [comment]
I didn't go deep into the architecture of the client. I am facting this issue as well. 
Device 1:
Pixel 7
Os: Graphene os
Device 2:
Iqoo Neo 7
OS: Fun-touch os ( Default from manufacturer)

Yesterday I received a mail at 3:22 I am currently writing this at 7:52. I didn't recieve a notification on either of my devices. 
I opened the client on Pixel this morning. I then sent an email from an alternate mail provider. It showed up on the pixel but not on the iqoo. 

So i as someone suggested that the notification service must be going to sleep after a certain period of time. 

Someone mentioned that this is not a messaging service and therefore the notification service must not be running all the time. With all due respect, I disagree. There are emails that are time sensitive and must be taken care of as Soon as One Receives it. 

Its it not acceptable if i recieve a mail at 3:22 and don't get notified on either of my devices for several hours it is worse if someone is paying for the service. Maybe give the user the option to run the notification service 24x7 or every 15 min instead of deciding for us.

**Reply by charlag** [reply]
Hi, the system should run the service at least every 15min, we just don't keep it running forever afterwards. If it does not work something is up with the system that is outside of our control. At least with battery optimizations disabled you should experience no issues, especially on GrapheneOS.
see https://dontkillmyapp.com/

**Reply by paridhi089** [reply]
I checked battery optimization has been disabled and autostart is enabled.

**Comment by BradB58** [comment]
Just now opened Tuta on my phone and it loaded in two new emails. One dated December 19 and one dated the 20th. No notifications received. I am attaching the log. From the time stamped in the log, it has been 31 hours since my device last booted. 

Device: Google Pixel 4a (5G)
OS: CalyxOS version 5.14.0
Android Version: 14

[1734832090005.txt](https://github.com/user-attachments/files/18220536/1734832090005.txt)

**Reply by charlag** [reply]
Hi, thank you for the log. There were multiple odd things in the logs.

```
12-21 19:33:54.149  7133  7195 D SSE     : Starting SSE connection
12-21 19:33:54.149  7133  7195 D SSE     : sse info not available skip reconnect
```

was it the first login? There should have been a push identifier on your device already

```
12-21 19:34:55.686  7133  7195 D SSE     : connection failed, MRHmmt0----2 okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL 200
12-21 19:34:55.689  7133  7195 E SSE     : error opening sse, rescheduling after 243, failedConnectionAttempts: 1
12-21 19:34:55.689  7133  7195 E SSE     : okhttp3.internal.http2.StreamResetException: stream was reset: CANCEL
```

the connection got interrupted pretty quickly which is odd but was restarted right after.

later once again
```
12-21 19:40:27.108  7133  7697 D SSE     : Starting SSE connection
12-21 19:40:27.108  7133  7697 D SSE     : sse info not available skip reconnect
```

are you deleting credentials every time?

**Reply by paridhi089** [reply]
how do i get the logs
